### PR TITLE
Piece movement

### PR DESCRIPTION
**Pull request by benoxoft**

### CODE REVIEW

File: Piece.cpp [hunk 1]
```diff
@@ -6,78 +6,132 @@
...
+    for ( int i = 0; i < board.sectors.size(); i++ ) {
+      for ( int r = 0; r < board.sectors[i].size(); r++ ) {
+        if ( board.sectors[r][i].contains(sf::Vector2f((float)event.mouseButton.x, (float)event.mouseButton.y)) ) {
+          if(validateMove(board, sf::Vector2i(i+1, r+1))){
+            sf::FloatRect snapRect = board.sectors[r][i];
+            sectorPosition=sf::Vector2i(i+1, r+1);
+            position.x = snapRect.left;
+            position.y = snapRect.top;
+            distributePosition();
+            goto foundPiece;
```
Comment (benoxoft) [inline Piece.cpp]: this goto could be rewritten as a method call

File: Piece.cpp [hunk 2]
```diff
@@ -6,78 +6,132 @@
...
+            for (auto m : moveset.offsets) {  
+              if ( pos == sectorPosition+m ) {
+                return true;
+              }
+            }
+  return false;
+}
+
+void Piece::onEvent(sf::Event event, Board &board) {
+  if ( beingMoved
```
Comment (benoxoft) [inline Piece.cpp:97]: reducing the imbrication levels would help make the code more readable.

maybe return early if the conditions are not met example:
```
if(false) {
  return;
}
// actual code
```

**Comment by benoxoft** [comment]
@Nopey @sweetsbeats I added 2 suggestions regarding the code everything else is 👍

**Comment by benoxoft** [comment]
👍

**Review by Nopey — APPROVED** [review]
teststate should probably be even more hands-off with the chesspiece's initialization

**Pull request merged**